cmd/settings-idp-set-saml.go

```diff
@@ -0,0 +1,130 @@
+// Package cmd implements access-cli commands
+package cmd
+
+/*
+Copyright © 2020 Barracuda Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	api "github.com/barracuda-cloudgen-access/access-cli/client/identity_providers"
+	"github.com/barracuda-cloudgen-access/access-cli/models"
+	"github.com/spf13/cobra"
+)
+
+// setSamlIdpCmd represents the get command
+var setSamlIdpCmd = &cobra.Command{
+	Use:   "saml",
+	Short: "Set SAML idp configuration",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		err := preRunCheckAuth(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		err = preRunFlagChecks(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		tw := identityProviderConfigBuildTableWriter()
+		createdList := []*models.IdentityProvider{}
+		total := 0
+
+		err := forAllInput(cmd, args, true,
+			func(values *inputEntry) (interface{}, error) { // do func
+				total++ // this is the total of successful+failures, must increment before failure
+				params := api.NewCreateIdentityProviderParams()
+
+				idp := &api.CreateIdentityProviderParamsBodyIdentityProvider{
+					IdpType: "saml",
+					Details: map[string]interface{}{},
+				}
+
+				err := placeInputValues(cmd, values, idp,
+					func(s string) { idp.Details["entity_id"] = s },
+					func(s string) { idp.Details["sso_url"] = s },
+					func(s string) { idp.Details["certificate"] = s })
```

Comment:
how will this work? multiline cert input is tricky

Reply:
should we read it from a file?

Reply:
yes, I think it should be "--cert_file" or something like that

Reply:
@pablosole added `--certificate-file` option to read from file.